packages/teleterm/src/ui/services/statePersistence/statePersistenceService.ts

```diff
@@ -58,8 +75,11 @@ export class StatePersistenceService {
       workspacesState: {
         workspaces: {},
       },
+      shareFeedback: {
+        hasBeenOpened: false,
+      },
     };
-    return this._fileStorage.get('state') || defaultState;
+    return { ...defaultState, ...this._fileStorage.get('state') };
```

Comment:
While the change here itself is fine IMHO, I'm commenting on this just because I want us to be super careful here. This encroaches on that "migrations for the persisted state" territory that we've talked about.

Any top-level additions to the default state should be safe but anything other than that is a breaking change, isn't it?

From my experience, once we'll be up against making a breaking change, it'll be much easier to add some hacks and defensive code inside the app code itself rather than build a storage system that can enforce that the saved state has certain fields, for example through migrations.

Reply:
>While the change here itself is fine IMHO, I'm commenting on this just because I want us to be super careful here. This encroaches on that "migrations for the persisted state" territory that we've talked about.

I agree, that we should be really careful here.

>Any top-level additions to the default state should be safe but anything other than that is a breaking change, isn't it?

Yes. I did this change to allow us to add new top-level sections.